src/docs/arch-session.dox

```diff
@@ -0,0 +1,53 @@
+/*! @arch_page arch-session Session
+
+@section arch_session_def Definition
+After a @ref arch-connection has been established between the application and WiredTiger, the
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
+connection. The maximum number of sessions is set through the configuration item \c session_max as
+part of the configuration string given in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user session". It exists another
+type of session called "internal session". Internal sessions are created and managed by WiredTiger
+internally and they are associated to global services at runtime such as @ref arch-eviction, @ref
+arch-checkpoint, @ref arch-rts, ...
+
+@section arch_session_ops Operations
+The different operations a session can request to WiredTiger are related to cursors handles, table
+operations and transactions. You can read the complete description of each possible operation in the
+documentation related to WT_SESSION.
+
+@section arch_session_txn Transactions
+It is possible to group a number of operations within a session, in other words, multiple operations
+can be treated as a single atomic operation. This can be done using @ref arch-transaction. When
+multiple operations are part of a transaction, they must either all be successful or fail;
+WiredTiger will abort any successful operation before a failure within the same transaction.
+Finally, one session can hold a unique transaction at any given time and this transaction only
+belongs to that same session.
```

Comment:
Can a session have multiple running transactions?

Reply:
No @AlexeyAnisimovAU, it cannot.

Reply:
Maybe we should mention that here?

Reply:
I think it is clear with the following sentence:
`Furthermore,
one session can hold a unique transaction at any given time and this transaction only belongs to
that session.`

Reply:
To me, `unique` doesn't imply that a session can have _only one_ running transaction. Maybe if we replace 'a unique' with `only one active` it would sound more clear? Something like this: 
`Furthermore, a session can hold only one running transaction at any given time and this transaction only belongs to that session.`
We need to make sure the reader understands that there's a one-to-one relationship.

Reply:
I will make the change, it is effectively clearer this way. Thanks !

Reply:
See 96500e52cc2c6e42d2507892f24a6b0c9b955371